tests/incremental/hashes/trait_defs.rs

```diff
@@ -30,7 +30,7 @@ trait TraitVisibility { }
 #[cfg(not(any(cfail1,cfail4)))]
 #[rustc_clean(cfg="cfail2")]
 #[rustc_clean(cfg="cfail3")]
-#[rustc_clean(cfg="cfail5", except="opt_hir_owner_nodes")]
+#[rustc_clean(cfg="cfail5", except="opt_hir_owner_nodes,predicates_of")]
```

Comment:
why are predicates_of now dirty?

Reply:
I'm not sure - it fails without the change. `predicates_of` is where the default bounds/supertrait are added, so I just assumed it was related to that.

Reply:
right, the question is why. I don't think your changes should have affected that, so some additional query edges seem to have been added. I wish incremental tests told us why something was dirty 🙃 Would be nice to fix or at least know why

Reply:
Confirmed that this change to the test is definitely still necessary.

My best guess is that I check `tcx.trait_is_auto` now? I need that to avoid adding the default supertrait to things like `Copy`, because if I do that, I can't relax it back with `PointeeSized` unless I do it with a `cfg(bootstrap)` (beta rustc won't allow explicit supertraits on auto traits). This could be fixed later by allowing the sizedness traits to be added to auto traits and then once that is in beta, removing the `trait_is_auto` condition (or doing that now with a `cfg` on all(?) our auto traits). 

Other than that, the only methods called in diff additions in all of `rustc_hir_analysis` on `tcx` are:

- `require_lang_item`
- `is_lang_item`
- `is_default_trait`
- `trait_is_auto`